app/controllers/lead_providers/report_schools_controller.rb

```diff
@@ -3,5 +3,24 @@
 module LeadProviders
   class ReportSchoolsController < ::LeadProviders::BaseController
     def start; end
+
+    def choose_delivery_partner
+      @delivery_partner_form = LeadProviderDeliveryPartnerForm.new
+      @delivery_partners = current_user.lead_provider.delivery_partners
+    end
+
+    def check_delivery_partner
+      @delivery_partner_form = LeadProviderDeliveryPartnerForm.new(
+        params.require(:lead_provider_delivery_partner_form).permit(:delivery_partner_id),
+      )
+
+      if @delivery_partner_form.valid?
+        session[:delivery_partner_id] = @delivery_partner_form.delivery_partner_id
```

Comment:
I would be tempted to make the form serializable and do `@delivery_partner_form.serializable_hash` here.

Reply:
Also, should we wipe the session in `start` action?

Reply:
Definitely agree with the comment about the session. I can't see any advantage to making the form serializable though. If there were many fields and / or I was using this to initialise an object then may be.

Reply:
My expectation is that we will use a single form model for this step and the next step of selecting schools, either via CSV or UI. The changes that requires would be part of the later work though

Reply:
@ethanmills I don't see how we can use a single form model for this part and the csv upload part. Those two things are just not related logically.